### ISSUE TITLE
Should restart docker daemon when new docker binary is put into place

### ISSUE THREAD

**Issue by jwm** [bflad/chef-docker#420]
The LWRP should restart the docker daemon when its `remote_file[docker_bin]` resource updates the docker binary on disk. Otherwise, invocations of the new docker binary fail when they try to use a newer version of the API to communicate with the running docker daemon.

I would submit a PR, but I'm having trouble figuring out how to have the DockerService LWRP to notify its own restart action in the `remote_file` resource. It seems like this should be a straightforward change, but if there's anything I can do to help or test, please let me know!

**Comment by jperville** [comment]
To give a concrete example,

For example, if my recipe had the following code:

```ruby
docker_service 'default' do
  # implicit version for ubuntu 14.04: docker 1.6.2
  action [:create, :start]
end
```

And converges successfully,
Then I change my code to:

```ruby
docker_service 'default' do
  version '1.8.1' # because I need a docker 1.8 feature such as fluentd logging drivers.
  action [:create, :start]
end
```

Then the node will appear to successfully converge, but any interactive docker command done after the chef-run will fail, for example this one:

```sh
root@docker2-ubuntu-1404:~# docker images
Error response from daemon: client and server don't have same version (client : 1.20, server: 1.18)
```

Upgrading the docker binary should have resulted in restarting the docker service, so that the client binary and the server always talk the same version of the API.

**Comment by someara** [comment]
Fixed in v1.0.34
